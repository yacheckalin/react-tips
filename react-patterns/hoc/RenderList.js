import React, { Component } from "react";
import axios from "axios";

export default function RenderUserList(WrappedComponent) {
  return class extends Component {
    constructor(props) {
      super(props);
      this.state = { data: [] };
    }
    async componentDidMount() {
      const { data } = await axios(
        "https://jsonplaceholder.typicode.com/users "
      );
      this.setState({ data });
    }
    render() {
      return <WrappedComponent data={this.state.data} {...this.props} />;
    }
  };
}

export const UserList = ({ data }) => (
  <ul className="collection">
    {data.map((item) => (
      <li key={item.id} className="collection-item">
        {item.name}
      </li>
    ))}
  </ul>
);
