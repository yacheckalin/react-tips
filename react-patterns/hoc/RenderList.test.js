import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import RenderList, { UserList } from "./RenderList";

describe("<RenderList />", () => {
  it("renders and displays properly", () => {
    const HOC = RenderList(UserList);
    const wrapper = shallow(<HOC />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
