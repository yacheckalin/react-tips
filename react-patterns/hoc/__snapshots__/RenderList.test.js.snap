// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderList /> renders and displays properly 1`] = `
<UserList
  data={Array []}
/>
`;
